extensions/ql-vscode/test/pure-tests/variant-analysis.test.ts

```diff
@@ -0,0 +1,12 @@
+import { expect } from 'chai';
+import { parseVariantAnalysisQueryLanguage, VariantAnalysisQueryLanguage } from '../../src/remote-queries/shared/variant-analysis';
+
+describe('parseVariantAnalysisQueryLanguage', () => {
```

Comment:
Maybe this function is too simple to require / benefit from a test. I'm happy to remove it if that would be simpler.

I also wasn't sure where to put it. The test is completely pure in that it doesn't need a workspace, and I have found other examples of tests in the `test/pure-tests` directory that test things from `src/remove-queries`. But I could move it back to `src/vscode-tests/no-workspace/remote-queries/shared/` if that would be a better location for it.

Reply:
I'm also not quite sure how the distinction between those tests was originally determined. However, I do think we should always prefer the `test` directory over `vscode-tests`; the former is much quicker to run and less flaky. We could also consider placing it in a directory like `test/remote-queries` and adding that to the `test:unit` `package.json` script.

Reply:
I think I'll leave it here for now unless you have strong thoughts on this. Moving it in the future will be very easy.